src/prefect/engine/task_runner.py

```diff
@@ -613,6 +616,34 @@ def load_results(
         """
         return state, upstream_states
 
+    def check_target(self, state: State, inputs: Dict[str, Result]) -> State:
+        """
+        Checks if a Result exists at the task's target.
+
+        Args:
+            - state (State): the current state of this task
+            - inputs (Dict[str, Result]): a dictionary of inputs whose keys correspond
+                to the task's `run()` arguments.
+
+        Returns:
+            - State: the state of the task after running the check
+        """
+        result = self.task.result
+        target = self.task.target
+
+        if result and target:
+            if result.exists(target):
+                cached_state = Cached(
+                    result=state._result,
+                    cached_inputs=inputs,
+                    cached_result_expiration=None,
+                    cached_parameters=prefect.context.get("parameters"),
+                    message=f"Result found at task target {target}",
```

Comment:
We can worry about this later, but because `target` can be a templated string we might want to adjust this message